islandora_object_lock_length/islandora_object_lock_length.module

```diff
@@ -56,18 +56,62 @@ function islandora_object_lock_length_manage_lock_form(array $form, array &$form
   $form_state['storage'] = array('pid' => $object->id);
   $global_lock_length = variable_get('islandora_object_lock_duration', 30);
   $collection_lock = islandora_object_lock_length_get_collection_lock($object->id);
-  return array(
-    'duration' => array(
-      '#type' => 'textfield',
-      '#title' => t('Collection level Lock duration (minutes)'),
-      '#default_value' => (isset($collection_lock->duration) ? $collection_lock->duration : $global_lock_length),
-      '#description' => t('Override default lock length for all children of this collection (objects and subcollections).'),
+  $default_duration = (isset($collection_lock->duration) ? $collection_lock->duration : $global_lock_length);
+  if ($default_duration == 'indefinite' || $default_duration == FALSE) {
```

Comment:
Should this strictly compare with `FALSE`? Other things like `0` or the empty string could trick this... Or is this by design?